builder/openstack/step_source_image_info.go

```diff
@@ -0,0 +1,100 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/gophercloud/gophercloud/pagination"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepSourceImageInfo struct {
+	SourceImage     string
+	SourceImageName string
+	ImageFilters    ImageFilterOptions
+}
+
+type ImageFilterOptions struct {
+	Filters    map[string]interface{} `mapstructure:"filters"`
+	MostRecent bool                   `mapstructure:"most_recent"`
+}
+
+func (s *StepSourceImageInfo) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	config := state.Get("config").(Config)
+	ui := state.Get("ui").(packer.Ui)
+
+	client, err := config.imageV2Client()
+
+	// if an ID is provided we skip the filter since that will return a single or no image
+	if s.SourceImage != "" {
+		state.Put("source_image", s.SourceImage)
+		return multistep.ActionContinue
+	}
+
+	params := &images.ListOpts{}
+
+	// build ListOpts from filters
+	if len(s.ImageFilters.Filters) > 0 {
+		errs := buildImageFilters(s.ImageFilters.Filters, params)
+		if len(errs.Errors) > 0 {
+			err := fmt.Errorf("Errors encountered in filter parsing.\n%s" + errs.Error())
+			state.Put("error", err)
+			ui.Error(errs.Error())
+			return multistep.ActionHalt
+		}
+	}
+
+	// override the "name" provided in filters if "s.SourceImageName" was filled
+	if s.SourceImageName != "" {
+		params.Name = s.SourceImageName
+	}
```

Comment:
`source_image_name` should override `source_image_filter` completely so remove this.

Reply:
To confirm, you would like the functionality to be to use only an image name if `source_image_name` is provided, even in the case where the configuration includes `source_image_filter` as well? This code allows current configs to be used as they are (image name kept where it is) and filters provided separately, for example.